libpod/container.go

```diff
@@ -407,6 +409,58 @@ func (c *Container) teardownStorage() error {
 	return nil
 }
 
+func (c *Container) addImageData(g *generate.Generator) (err error) {
+	var (
+		args []string
+	)
+	if !c.config.UseImageConfig {
+		return nil
+	}
+	imageData, err := c.runtime.GetImageConfig(c.config.RootfsImageID)
+	if err != nil {
+		return errors.Wrapf(err, "error reading container image data")
+	}
+	if imageData.User != "" {
+		uid, _, err := chrootuser.LookupUserInContainer(c.bundlePath(), imageData.User)
+		if err != nil {
+			return errors.Wrapf(err, "user %q is not defined in %q", imageData.User, c.bundlePath())
+		}
+		g.SetProcessUID(uint32(uid))
+	}
+	for _, e := range imageData.ExposedPorts {
+		//TODO exposed ports are not handled yet
+		logrus.Warn("Exposed port %q is ignored", e)
+	}
+
+	if len(imageData.Cmd) > 0 {
+		args = imageData.Cmd
+	}
+	if len(imageData.Entrypoint) > 0 {
+		args = imageData.Entrypoint
+	}
+	if len(args) > 0 {
+		g.SetProcessArgs(args)
```

Comment:
I think we only set these if the user hasn't already set args to override

Reply:
Yes I agree.  The problem is it is not easy to tell.  It would be easier if this function could be called before we process the user data.